python/rmm/rmm/_lib/memory_resource.pyx

```diff
@@ -366,6 +378,49 @@ cdef class ManagedMemoryResource(DeviceMemoryResource):
         pass
 
 
+cdef class SystemMemoryResource(DeviceMemoryResource):
+    def __cinit__(self):
+        self.c_obj.reset(
+            new system_memory_resource()
+        )
+
+    def __init__(self):
+        """
+        Memory resource that uses ``malloc``/``free`` for
+        allocation/deallocation.
+        """
+        pass
+
+
+cdef class SamHeadroomResourceAdaptor(DeviceMemoryResource):
```

Comment:
An adaptor accepts another resource as an argument. This is not an adaptor, it is implemented like an entirely new resource.

Reply:
Should we redesign this so that the SystemMemoryResource has an argument `headroom` and we don't need the "adaptor" at all? Or rename this so it's not an adaptor?

Reply:
The C++ side has both the system mr and the adaptor, but I guess we don't have to strictly mirror it in python. I'd be happy to have a single SystemMemoryResource with an argument.

Reply:
Done.